### PR TITLE
Ensure that the most precise error information is included in a scan run result

### PR DESCRIPTION
**Pull request by alisonlomaka**
#### Details
A previous change serialized a BrowserError to string before passing it into Runner.setRunResult. This had the unintended side effect of resulting in all BrowserErrors being reported as InternalError, instead of being mapped to the corresponding ScanRunErrorCode.

This change explicitly converts Error and BrowserError objects into ScanError objects before passing them to setRunResult. This works because both BrowserError and ScanError objects have an errorType member and the BrowserErrorTypes and ScanErrorTypes align. Error has no overlap with ScanError so it is mapped to InternalError - which is what the conversion handling would ultimately do with a string anyway.

For reference see also ScanResponseConverter.createFailedScanResponse and ScanErrorConverter.getScanRunErrorCode; this is the code path whereby a string | ScanError is converted into a ScanRunError for inclusion in a ScanResultResponse.

##### Motivation
Addresses internal bug 2007185 written against WCP.

##### Context
This was working before by a lucky loophole. We were passing a variable of type string | BrowserError into a type of string | ScanError. The string types on both sides allowed it to pass compile but then sneak through a BrowserError at runtime, which happened to be assignable and consumable as a ScanError.

I don't love this solution but it does make the luck and assumptions explicit, and it is narrowly scoped. Any change to the actual types defined for OnDemandPageScanResult or ScanResultResponse would have had a much wider ripple effect including the potential to affect a11y as well.

#### Pull request checklist

- [x] Addresses an existing issue: Fixes #0000
- [n/a] Added relevant unit test for your changes. (`yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [ ] Ran precheckin (`yarn precheckin`)
- [x] Validated in an Azure resource group

### CODE REVIEW

File: packages/privacy-scan-runner/src/runner/runner.spec.ts [hunk 1]
```diff
@@ -244,10 +246,14 @@ function setupProcessScanResult(): void {
         if ((privacyScanResults.error as BrowserError)?.errorType === 'BannerXPathNotDetected') {
             runState = pageScanResult.run?.retryCount >= maxFailedScanRetryCount ? 'completed' : 'failed';
         }
+        const scanError = privacyScanResults.error as ScanError;
+        if (isEmpty(scanError.errorType)) {
+            scanError.errorType = 'InternalError';
+        }
```
Comment (pownkel) [inline packages/privacy-scan-runner/src/runner/runner.spec.ts:252]: Nit (errorType will be overwritten if it exists in privacyScanResults.error)
```suggestion
        const scanError: ScanError = {
            errorType: 'InternalError'
            ...privacyScanResults.error,
        };
```

File: packages/privacy-scan-runner/src/runner/runner.spec.ts [hunk 2]
```diff
@@ -244,10 +246,14 @@ function setupProcessScanResult(): void {
         if ((privacyScanResults.error as BrowserError)?.errorType === 'BannerXPathNotDetected') {
             runState = pageScanResult.run?.retryCount >= maxFailedScanRetryCount ? 'completed' : 'failed';
         }
+        const scanError = privacyScanResults.error as ScanError;
+        if (isEmpty(scanError.errorType)) {
+            scanError.errorType = 'InternalError';
```
Comment (pownkel) [inline packages/privacy-scan-runner/src/runner/runner.spec.ts:251]: Is it still possible for privacyScanResults.error to be a string? What happens here in that case?

File: packages/privacy-scan-runner/src/runner/runner.ts [hunk 3]
```diff
@@ -186,6 +186,15 @@ export class Runner {
         return !isEmpty(reports) ? this.reportWriter.writeBatch(reports) : undefined;
     }
 
+    private convertToScanError(error: Error | BrowserError): ScanError {
+        const scanError = error as ScanError;
+        if (isEmpty(scanError.errorType)) {
+            scanError.errorType = 'InternalError';
+        }
```
Comment (pownkel) [inline packages/privacy-scan-runner/src/runner/runner.ts:193]: Nit
```suggestion
        const scanError: ScanError = {
            errorType: 'InternalError',
            ...error,
        };
```

File: packages/privacy-scan-runner/src/runner/runner.spec.ts (hunk 2 above)
Comment (alisonlomaka) [reply packages/privacy-scan-runner/src/runner/runner.spec.ts:251]: It is not; Maxim's previous changes modified the signature for PrivacyScanResult.error to be Error | BrowserError.

**Review by pownkel — APPROVED** [review]

**Pull request merged**